app/models/study_file.rb

```diff
@@ -1331,6 +1333,45 @@ def set_cluster_group_ranges
     end
   end
 
+  # update clustering fragments to be called on updates to study_file
+  def call_ingest_on_updated_clustering_fragments
```

Comment:
The method name seems pretty long (though descriptive).  Maybe `handle_data_fragment_updates` since some are just database calls and others new parse jobs?

Reply:
Since this method (currently) only does stuff on clustering fragments I'll go with `handle_clustering_fragment_updates`